### PR TITLE
Don't send empty attribute list in SignUp/start

### PR DESCRIPTION
**Pull request by nilo-ms**
## Proposed changes

Before this PR we were using an empty attribute list as default value when the user didn't specify any attribute. This resulted in adding an empty list in the signUp/start HTTP call

## Type of change

- [ ] Feature work
- [X] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [X] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

## Additional information
I considered other methods for resolving this problem as well. I could just modify the `MSALNativeAuthSignUpRequestProvider` class and add the attributes field in the HTTP request only when the list was not empty. I didn't think this was the right way to solve the issue, because the RequestProvider class should provide the request based on the input parameters, not based to internal logic. So, I used a nil value instead of empty list as default attributes value in the PublicClientApplication class.

### CODE REVIEW

File: MSAL/src/native_auth/network/sign_up/MSALNativeAuthSignUpRequestProvider.swift [hunk 1]
```diff
@@ -92,7 +89,10 @@ final class MSALNativeAuthSignUpRequestProvider: MSALNativeAuthSignUpRequestProv
         return request
     }
 
-    private func formatAttributes(_ attributes: [String: Any]) throws -> String? {
+    private func formatAttributes(_ attributes: [String: Any]?) throws -> String? {
+        guard let attributes = attributes else {
```
Comment (spetrescu84) [inline MSAL/src/native_auth/network/sign_up/MSALNativeAuthSignUpRequestProvider.swift:93]: Should we strip them also if the developer passes an empty dictionary? As in calls public client interface like this?
`signUpUsingPassword(username: "Test", password: "Pass", attributes: [:], delegate: self)` 
That would mean doing `guard let attributes = attributes, !attributes.isEmpty else {` here

File: MSAL/test/unit/native_auth/public/MSALNativeAuthPublicClientApplicationTest.swift [hunk 2]
```diff
@@ -76,6 +76,7 @@ final class MSALNativeAuthPublicClientApplicationTest: XCTestCase {
         let delegate = SignUpPasswordStartDelegateSpy(expectation: exp)
         sut.signUpUsingPassword(username: "", password: "", delegate: delegate)
         wait(for: [exp], timeout: 1)
+        XCTAssertNil(controllerFactoryMock.signUpController.signUpStartRequestParameters?.attributes)
```
Comment (spetrescu84) [inline MSAL/test/unit/native_auth/public/MSALNativeAuthPublicClientApplicationTest.swift:79]: Do you think these checks are needed for these specific tests? The test names are not describing anything around this. They don't hurt to be there but I think they are a bit overkill. We could have a test where we check that if we don't set the attributes they are not sent in this class but I think it should be separate.

File: MSAL/src/native_auth/network/sign_up/MSALNativeAuthSignUpRequestProvider.swift (hunk 1 above)
Comment (nilo-ms) [reply MSAL/src/native_auth/network/sign_up/MSALNativeAuthSignUpRequestProvider.swift:93]: Hey Silviu, I took in consideration this approach. Please check the PR description to know more

File: MSAL/test/unit/native_auth/public/MSALNativeAuthPublicClientApplicationTest.swift (hunk 2 above)
Comment (nilo-ms) [reply MSAL/test/unit/native_auth/public/MSALNativeAuthPublicClientApplicationTest.swift:79]: Yeah, actually make more sense to have a dedicated test for this, so we can highlight this behaviour. Thanks I'll update this test class

**Pull request closed without merging**